applications/StructuralMechanicsApplication/custom_elements/small_displacement_mixed_volumetric_strain_element.cpp

```diff
@@ -965,9 +908,29 @@ int  SmallDisplacementMixedVolumetricStrainElement::Check(const ProcessInfo& rCu
 {
     KRATOS_TRY
 
-    int base_element_check = SmallDisplacementMixedVolumetricStrainElement::BaseType::Check(rCurrentProcessInfo);
+    int check = SmallDisplacementMixedVolumetricStrainElement::BaseType::Check(rCurrentProcessInfo);
+
+    // Base check
+    check = StructuralMechanicsElementUtilities::BaseElementCheck(this, rCurrentProcessInfo);
+
+    // Verify that the variables are correctly initialized
+    KRATOS_CHECK_VARIABLE_KEY(THICKNESS)
+
+    // Check that the element's nodes contain all required SolutionStepData and Degrees of freedom
+    const auto& r_geometry = this->GetGeometry();
+    for ( IndexType i = 0; i < r_geometry.size(); i++ ) {
+        const NodeType& r_node = r_geometry[i];
+        KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(VOLUMETRIC_STRAIN,r_node)
+
+        KRATOS_CHECK_DOF_IN_NODE(VOLUMETRIC_STRAIN, r_node)
+    }
+
+    // Check constitutive law
+    if ( mConstitutiveLawVector.size() > 0 ) {
+        return mConstitutiveLawVector[0]->Check( GetProperties(), GetGeometry(), rCurrentProcessInfo );
```

Comment:
Maybe it is a good idea that we also put this in the `BaseElementCheck`.

Reply:
Ok